kafka/integration_test.go

```diff
@@ -3245,6 +3245,95 @@ func (its *IntegrationTestSuite) TestAdminClient_ListOffsets() {
+	assert.Nil(createTopicError, "Create Topics should not fail.")
+	assert.Equal(createTopicResult[0].Error.Code(), ErrNoError, "Create Topics Error Code should be ErrNoError.")
+
+	p, err := NewProducer(&ConfigMap{"bootstrap.servers": bootstrapServers})
+	assert.Nil(err, "Unable to create Producer.")
+	defer p.Close()
+
+	numMessages := 3
+	numPartitions := 3
+	for i := 0; i < numMessages; i++ {
+		for j := 0; j < numPartitions; j++ {
+			p.Produce(&Message{
+				TopicPartition: TopicPartition{Topic: &topic, Partition: int32(j)},
+				Value:          []byte(fmt.Sprintf("Message-%d", i)),
+			}, nil)
+		}
+	}
+
+	p.Flush(5 * 1000)
+
+	// Delete the records:
+	// Partition 0: upto offset 2
+	// Partition 1: upto the last offset
+	// Partition 2: upto offset 4 (non-existent)
+	// The result will contain the minimum offset available after deletion in that Topic Partiton
+	delRecordsTopicPartitionOffsets := []TopicPartition{
+		{Topic: &topic, Partition: int32(0), Offset: Offset(2)},
+		{Topic: &topic, Partition: int32(1), Offset: OffsetEnd},
+		{Topic: &topic, Partition: int32(2), Offset: Offset(4)},
+	}
+	deleteRes, err := a.DeleteRecords(ctx, delRecordsTopicPartitionOffsets)
+	assert.Nil(err, "Delete Records should not fail")
+	assert.Len(deleteRes.DeleteRecordsResults, 3, "Length of deleteRes.TopicPartitions should be 3")
+
+	assert.Nil(deleteRes.DeleteRecordsResults[0].TopicPartition.Error,
+		"Error should not be set in deleteRes.DeleteRecordsResults[0]")
+	assert.Nil(deleteRes.DeleteRecordsResults[1].TopicPartition.Error,
+		"Error should not be set in deleteRes.DeleteRecordsResults[1]")
+	assert.Error(deleteRes.DeleteRecordsResults[2].TopicPartition.Error,
+		"Error should be set in deleteRes.DeleteRecordsResults[2]")
+	assert.Nil(deleteRes.DeleteRecordsResults[2].DeletedRecords,
+		"DeletedRecords should be nil within deleteRes.DeleteRecordsResults[2]")
+
+	// Offsets after deletion reported by DeleteRecords, in order of topic partitions.
+	offsetAfterDeletion := []Offset{
+		deleteRes.DeleteRecordsResults[0].DeletedRecords.LowWatermark,
+		deleteRes.DeleteRecordsResults[1].DeletedRecords.LowWatermark,
+	}
+
+	// Find the minimum offsets in the partitions of the topic via ListOffsets.
+	// It should be equal to the offset we get after the deletion operation.
+	topicPartitionOffsets := map[TopicPartition]OffsetSpec{
+		{Topic: &topic, Partition: 0}: EarliestOffsetSpec,
+		{Topic: &topic, Partition: 1}: EarliestOffsetSpec,
+	}
+
+	results, err := a.ListOffsets(ctx, topicPartitionOffsets)
+	assert.Nil(err, "ListOffsets should not fail.")
+	assert.Len(results.ResultInfos, 2, "Length of results.ResultInfos should be 2")
+
+	for toppar, info := range results.ResultInfos {
+		assert.Equal(info.Error.Code(), ErrNoError, "Error code should be ErrNoError.")
+		assert.Equal(info.Offset, offsetAfterDeletion[toppar.Partition], "Offset should be equal to the offset obtained after deletion.")
```

Comment:
Line break only
```suggestion
		assert.Equal(info.Offset, offsetAfterDeletion[toppar.Partition],
			"Offset should be equal to the offset obtained after deletion.")
```

Reply:
Let's add a verification of the expected offset, not only that it equals to ListOffsets result
```go
		switch toppar.Partition {
		case 0:
			assert.Equal(info.Offset, Offset(2),
				"Offset should be equal to 2 on partition 0, got %v",
				info.Offset)
		case 1:
			assert.Equal(info.Offset, Offset(3),
				"Offset should be equal to 3 on partition 1, got %v",
				info.Offset)
		}
```

Reply:
Please merge master into here and recreate kafka/api.html to add `DeleteRecords` documentation.

Reply:
Done.
I have used a slightly different style within the loop to check the offsets we manually expect. And I included the third partition also within that check since we know that it will be at Offset(0)